### PR TITLE
update fixed change for vl_ltr

### PR DESCRIPTION
**Pull request by psky1111**
following the requirements and finish adjustments

### CODE REVIEW

File: ppcls/arch/backbone/model_zoo/vl_ltr.py [hunk 1]
```diff
@@ -0,0 +1,652 @@
+import os.path as osp
+import os
+from typing import Tuple, Union
+#check
+import paddle.distributed as dist
```
Comment (TingquanGao) [inline ppcls/arch/backbone/model_zoo/vl_ltr.py]: 这个没有用到的话，就去掉吧。

File: ppcls/arch/backbone/__init__.py [hunk 2]
```diff
@@ -88,6 +88,7 @@
 from .model_zoo.adaface_ir_net import AdaFace_IR_18, AdaFace_IR_34, AdaFace_IR_50, AdaFace_IR_101, AdaFace_IR_152, AdaFace_IR_SE_50, AdaFace_IR_SE_101, AdaFace_IR_SE_152, AdaFace_IR_SE_200
 from .model_zoo.wideresnet import WideResNet
 from .model_zoo.uniformer import UniFormer_small, UniFormer_small_plus, UniFormer_small_plus_dim64, UniFormer_base, UniFormer_base_ls
+from .model_zoo.vl_ltr import CVLP_vit16,LGR_vit16
```
Comment (TingquanGao) [inline ppcls/arch/backbone/__init__.py]: code style还是有问题，正常应该是“CVLP_vit16, LGR_vit16”，少了一个空格，建议再检查一下pre-commit

File: ppcls/arch/backbone/model_zoo/vl_ltr.py [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+            print(info)
+
+    def build_attention_mask(self):
+        mask = paddle.empty((self.context_length, self.context_length))
+        mask = paddle.full_like(mask, float("-inf"))
+        mask = paddle.triu(mask, diagonal=1)
+        return mask
+
+    @property
+    def dtype(self):
```
Comment (TingquanGao) [inline ppcls/arch/backbone/model_zoo/vl_ltr.py]: 这个是在其他地方会用到吗？如果后续使用fp16训练的话，这样写可能会有问题。

File: ppcls/arch/backbone/model_zoo/vl_ltr.py [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+            print('freeze attn norm')
+            for m in self.text_block.attn.parameters():
+                if isinstance(m, nn.LayerNorm):
+                    m.eval()
+                if isinstance(m, nn.BatchNorm2D):
+                    m.eval()
+        self.initialize_parameters()
+
+    @property
+    def dtype(self):
```
Comment (TingquanGao) [inline ppcls/arch/backbone/model_zoo/vl_ltr.py]: 同上

File: ppcls/configs/VL_LTR/VL_LTR_finetune.yaml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  save_inference_dir: ./inference
+
+# model architecture
+Arch:
+  name: LGR_vit16
+  class_num: 1000
+  context_length: 75
+  sent_length: 64
+  cache_root: "cached"
+  pretrained_clip: "vit"
```
Comment (TingquanGao) [inline ppcls/configs/VL_LTR/VL_LTR_finetune.yaml:26]: 这里是什么意思？

File: ppcls/data/dataloader/imagenetLT_dataset.py [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+    'graffiti of the {}.',
+    'a toy {}.',
+    'itap of my {}.',
+    'a photo of a cool {}.',
+    'a photo of a small {}.',
+    'a tattoo of the {}.',
+]
+"""
+CLIP text encoder and decoder
+"""
```
Comment (TingquanGao) [inline ppcls/data/dataloader/imagenetLT_dataset.py]: 这个注释是指的那里？

File: ppcls/data/preprocess/batch_ops/batch_operators.py [hunk 7]
```diff
@@ -65,11 +65,12 @@ class MixupOperator(BatchOperator):
 
     """
 
-    def __init__(self, class_num, alpha: float=1.):
+    def __init__(self, class_num, alpha: float=1., is_batch=False):
```
Comment (TingquanGao) [inline ppcls/data/preprocess/batch_ops/batch_operators.py]: 感觉is_batch不太好，是不是可以换个变量名？

File: ppcls/loss/vltloss.py
Comment (TingquanGao) [inline ppcls/loss/vltloss.py:1]: 这里面的几个类都用到了吗？cross_entropy和label smooth相关的代码是否可以复用ppcls/loss/celoss.py中的？

**Review by TingquanGao — COMMENTED** [review]
还有一些问题，麻烦再看下

**Comment by psky1111** [comment]
好的,我马上看下




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleClas"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月26日(星期一) 下午5:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleClas] update fixed change for vl_ltr (PR #2828)





 
@TingquanGao commented on this pull request.
 
还有一些问题，麻烦再看下

 
In ppcls/arch/backbone/model_zoo/vl_ltr.py:
 &gt; @@ -0,0 +1,652 @@ +import os.path as osp +import os +from typing import Tuple, Union +#check +import paddle.distributed as dist  
这个没有用到的话，就去掉吧。
 
 
In ppcls/arch/backbone/__init__.py:
 &gt; @@ -88,6 +88,7 @@  from .model_zoo.adaface_ir_net import AdaFace_IR_18, AdaFace_IR_34, AdaFace_IR_50, AdaFace_IR_101, AdaFace_IR_152, AdaFace_IR_SE_50, AdaFace_IR_SE_101, AdaFace_IR_SE_152, AdaFace_IR_SE_200  from .model_zoo.wideresnet import WideResNet  from .model_zoo.uniformer import UniFormer_small, UniFormer_small_plus, UniFormer_small_plus_dim64, UniFormer_base, UniFormer_base_ls +from .model_zoo.vl_ltr import CVLP_vit16,LGR_vit16  
code style还是有问题，正常应该是“CVLP_vit16, LGR_vit16”，少了一个空格，建议再检查一下pre-commit
 
 
In ppcls/arch/backbone/model_zoo/vl_ltr.py:
 &gt; +                    new_size, +                    num_extra_tokens=num_extra_tokens) +                pretrained_state_dict['pos_embed'] = new_pos_embed + +            info = self.set_state_dict(pretrained_state_dict) +            print('loaded pretrained clip.') +            print(info) + +    def build_attention_mask(self): +        mask = paddle.empty((self.context_length, self.context_length)) +        mask = paddle.full_like(mask, float("-inf")) +        mask = paddle.triu(mask, diagonal=1) +        return mask + +    @property +    def dtype(self):  
这个是在其他地方会用到吗？如果后续使用fp16训练的话，这样写可能会有问题。
 
 
In ppcls/arch/backbone/model_zoo/vl_ltr.py:
 &gt; +            for m in self.visual.parameters(): +                if isinstance(m, nn.LayerNorm): +                    m.eval() +                if isinstance(m, nn.BatchNorm2D): +                    m.eval() +        if self.attn_grad is False: +            print('freeze attn norm') +            for m in self.text_block.attn.parameters(): +                if isinstance(m, nn.LayerNorm): +                    m.eval() +                if isinstance(m, nn.BatchNorm2D): +                    m.eval() +        self.initialize_parameters() + +    @property +    def dtype(self):  
同上
 
 
In ppcls/configs/VL_LTR/VL_LTR_finetune.yaml:
 &gt; +  cache_root: "cached" +  pretrained_clip: "vit"  
这里是什么意思？
 
 
In ppcls/data/dataloader/imagenetLT_dataset.py:
 &gt; +""" +CLIP text encoder and decoder +"""  
这个注释是指的那里？
 
 
In ppcls/data/preprocess/batch_ops/batch_operators.py:
 &gt; @@ -65,11 +65,12 @@ class MixupOperator(BatchOperator):        """   -    def __init__(self, class_num, alpha: float=1.): +    def __init__(self, class_num, alpha: float=1., is_batch=False):  
感觉is_batch不太好，是不是可以换个变量名？
 
 
On ppcls/loss/vltloss.py:
 
这里面的几个类都用到了吗？cross_entropy和label smooth相关的代码是否可以复用ppcls/loss/celoss.py中的？

File: ppcls/arch/backbone/model_zoo/vl_ltr.py (hunk 1 above)
Comment (psky1111) [reply ppcls/arch/backbone/model_zoo/vl_ltr.py]: 在最新commit中已经去掉冗余部分

File: ppcls/arch/backbone/__init__.py (hunk 2 above)
Comment (psky1111) [reply ppcls/arch/backbone/__init__.py]: 在最新commit中修复

File: ppcls/arch/backbone/model_zoo/vl_ltr.py (hunk 4 above)
Comment (psky1111) [reply ppcls/arch/backbone/model_zoo/vl_ltr.py]: 已经去掉,使用默认dtype

File: ppcls/configs/VL_LTR/VL_LTR_finetune.yaml (hunk 5 above)
Comment (psky1111) [reply ppcls/configs/VL_LTR/VL_LTR_finetune.yaml:26]: 已经在相关yaml文件中加入注释. 这里cache_root是在训练时保存的token缓存. 会自动生成"cached/"这个目录.

File: ppcls/data/dataloader/imagenetLT_dataset.py (hunk 6 above)
Comment (psky1111) [reply ppcls/data/dataloader/imagenetLT_dataset.py]: 已修改注释,将其放到相应位置

File: ppcls/data/preprocess/batch_ops/batch_operators.py (hunk 7 above)
Comment (psky1111) [reply ppcls/data/preprocess/batch_ops/batch_operators.py]: 已经更改变量名字

File: ppcls/loss/vltloss.py
Comment (psky1111) [reply ppcls/loss/vltloss.py:1]: 删除loss文件中冗余的cross_entropy以及kl_div, 剩余的无法直接复用celoss,所以保留了

File: ppcls/arch/backbone/model_zoo/vl_ltr.py (hunk 3 above)
Comment (psky1111) [reply ppcls/arch/backbone/model_zoo/vl_ltr.py]: 已经去掉,使用默认函数
